cruise-control-client/cruisecontrolclient/client/CCParameter/SetOfChoicesParameter.py

```diff
@@ -0,0 +1,78 @@
+from typing import Set, Union, Collection  # noqa
+
+# To allow for warnings that don't break the program.
+#
+# Specifically, we don't want this client to break forward compatibility with
+# new cruise-control features and parameter choices.
+#
+# Accordingly, for enums, we want to warn, but not raise an Exception, when
+# a provided value is outside of that enum
+from warnings import warn
+
+from cruisecontrolclient.client.CCParameter.Parameter import AbstractParameter
+
+
+class AbstractSetOfChoicesParameter(AbstractParameter):
+    lowercase_set_of_choices: Set[str] = set()
+
+    def __init__(self, value: Union[str, Collection[str]]):
+        if isinstance(value, Collection) and type(value) != str:
+            value = ','.join(value)
+        AbstractParameter.__init__(self, value)
+
+    def validate_value(self):
+        if type(self.value) == str:
+            for i in self.value.split(','):
+                # Assume that cruise-control uses case-insensitive enums
+                if i.lower() not in self.lowercase_set_of_choices:
+                    warn(f"{i} is not in the valid set of choices")
+        else:
+            raise ValueError(f"{self.value} must be either a string or a Collection")
+
+
+class DisableSelfHealingForParameter(AbstractSetOfChoicesParameter):
+    """disable_self_healing_for=[Set-of-{@link AnomalyType}]"""
+    lowercase_set_of_choices = {'goal_violation', 'metric_anomaly', 'broker_failure'}
+    name = 'disable_self_healing_for'
+    description = "The anomaly detectors for which to disable self-healing"
+    argparse_properties = {
+        'args': ('--disable-self-healing-for', '--disable-self-healing'),
+        # nargs='+' allows for multiple of the set to be specified
+        'kwargs': dict(help=description, metavar='ANOMALY_DETECTOR', choices=lowercase_set_of_choices, nargs='+')
+
+    }
+
+
+class EnableSelfHealingForParameter(AbstractSetOfChoicesParameter):
+    """enable_self_healing_for=[Set-of-{@link AnomalyType}]"""
+    lowercase_set_of_choices = {'goal_violation', 'metric_anomaly', 'broker_failure'}
```

Comment:
I have noticed this; however unlike the Endpoint from `migrate_to_kafka_2_0` (which carries a warning about backwards-incompatibility), I don't have a great way of communicating to users that the `disk_failure` anomaly is only for `migrate_to_kafka_2_0` branch.

I feel like this is defensible, (I'd rather not add `disk_failure` yet because it's not supported) but it's arguably inconsistent.
I can easily add `disk_failure` anomaly if that resolves this inconsistency.

Reply:
Thanks for the clarification. If the existence of `disk_failure` in `migrate_to_kafka_2_0` branch does not break the client, I will merge this patch to `master`, and cherry-pick it in `migrate_to_kafka_2_0`.

Then we can create a separate patch to add support for `disk_failure` in `migrate_to_kafka_2_0` branch. If that sounds reasonable, would you mind creating an issue for this (for tracking)?

Reply:
-- Based on the offline discussion with @mgrubent, we agreed on the approach above.